test/Microsoft.AspNetCore.Mvc.TestCommon/TestFileProvider.cs

```diff
@@ -46,7 +46,7 @@ public TestFileInfo AddFile(string path, string contents)
             var fileInfo = new TestFileInfo
             {
                 Content = contents,
-                PhysicalPath = Path.Combine(Root, path),
+                PhysicalPath = Path.Combine(Root, NormalizeAndEnsureValidPhysicalPath(path)),
```

Comment:
`Path.GetFullPath(Path.Combine(Root, path))` would do no?

Reply:
If the path starts with a slash, Path.Combine would ignore the Root